cmd/antrea-agent/options.go

```diff
@@ -125,7 +126,12 @@ func (o *Options) validate(args []string) error {
 	}
 
 	if encapMode.SupportsNoEncap() {
-		if !features.DefaultFeatureGate.Enabled(features.AntreaProxy) {
+		// When use NoEncap traffic mode without Antrea Proxy, Pod-to-Service traffic is handled by iptables/ipvs in
+		// root netns, if the endpoint is not local the DNATed traffic will be output to physical network directly
+		// without going back to OVS for Egress NetworkPolicy enforcement, which breaks basic security functionality.
+		// It is not allowed to use in principle, But can force support NoEncap with ALLOW_NO_ENCAP_WITHOUT_ANTREA_PROXY
+		// environment parameter for performance.
```

Comment:
Some corrections / suggestions:

> When using NoEncap traffic mode without AntreaProxy, Pod-to-Service traffic is handled by kube-proxy (iptables/ipvs) in the root netns. If the Endpoint is not local the DNATed traffic will be output to the physical network directly without going back to OVS for Egress NetworkPolicy enforcement, which breaks basic security functionality. Therefore, we usually do not allow the NoEncap traffic mode without AntreaProxy. But one can bypass this check and force this feature combination to be allowed, by defining the ALLOW_NO_ENCAP_WITHOUT_ANTREA_PROXY environment variable and setting it to true. This may lead to better performance when using NoEncap if Egress NetworkPolicy enforcement is not required.

Reply:
Thanks for the review, updated.